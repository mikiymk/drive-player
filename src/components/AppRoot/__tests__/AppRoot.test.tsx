import { render } from "solid-js/web";
import { expect, test, vi } from "vitest";

import { MusicPlayer } from "../index";

test("1 second", () => {
  window.AudioContext = vi.fn();
  window.fetch = vi.fn(() =>
    Promise.resolve({
      async json() {
        /* eslint-disable camelcase */
        return {
          access_token: "access token",
          expires_in: 3599,
          refresh_token: "refresh token",
          scope: "scope",
          token_type: "token type",
          files: [],
          nextPageToken: undefined,
        };
      },
    } as Response),
  );

  const root = document.createElement("div");
  render(MusicPlayer, root);

  expect(root).toMatchSnapshot();
});
