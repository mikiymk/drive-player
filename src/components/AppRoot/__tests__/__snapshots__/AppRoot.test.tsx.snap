// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot app 1`] = `
<div>
  <div
    class="style_stylePlayer__12rw3ob9 style_themeClass__12rw3ob0"
  >
    <ul
      class="style_tab__1bi6wwc0"
    >
      <li
        class="style_tabItem__1bi6wwc1 style_tabItemSelected__1bi6wwc4"
      >
        <button
          class="style_tabItemButton__1bi6wwc2"
          type="button"
        >
          <svg
            class="style_icon__7h3wg50"
            viewBox="0 0 24 24"
          >
            <title>
              play
            </title>
            <path
              d="M8,5.14V19.14L19,12.14L8,5.14Z"
            />
          </svg>
          <span
            class="style_tabItemLabel__1bi6wwc3"
          >
            Now Playing
          </span>
        </button>
      </li>
      <li
        class="style_tabItem__1bi6wwc1"
      >
        <button
          class="style_tabItemButton__1bi6wwc2"
          type="button"
        >
          <svg
            class="style_icon__7h3wg50"
            viewBox="0 0 24 24"
          >
            <title>
              library
            </title>
            <path
              d="M7,5H21V7H7V5M7,13V11H21V13H7M4,4.5A1.5,1.5 0 0,1 5.5,6A1.5,1.5 0 0,1 4,7.5A1.5,1.5 0 0,1 2.5,6A1.5,1.5 0 0,1 4,4.5M4,10.5A1.5,1.5 0 0,1 5.5,12A1.5,1.5 0 0,1 4,13.5A1.5,1.5 0 0,1 2.5,12A1.5,1.5 0 0,1 4,10.5M7,19V17H21V19H7M4,16.5A1.5,1.5 0 0,1 5.5,18A1.5,1.5 0 0,1 4,19.5A1.5,1.5 0 0,1 2.5,18A1.5,1.5 0 0,1 4,16.5Z"
            />
          </svg>
          <span
            class="style_tabItemLabel__1bi6wwc3"
          >
            Library
          </span>
        </button>
      </li>
      <li
        class="style_tabItem__1bi6wwc1"
      >
        <button
          class="style_tabItemButton__1bi6wwc2"
          type="button"
        >
          <svg
            class="style_icon__7h3wg50"
            viewBox="0 0 24 24"
          >
            <title>
              playlist
            </title>
            <path
              d="M15,6H3V8H15V6M15,10H3V12H15V10M3,16H11V14H3V16M17,6V14.18C16.69,14.07 16.35,14 16,14A3,3 0 0,0 13,17A3,3 0 0,0 16,20A3,3 0 0,0 19,17V8H22V6H17Z"
            />
          </svg>
          <span
            class="style_tabItemLabel__1bi6wwc3"
          >
            Playlist
          </span>
        </button>
      </li>
      <li
        class="style_tabItem__1bi6wwc1"
      >
        <button
          class="style_tabItemButton__1bi6wwc2"
          type="button"
        >
          <svg
            class="style_icon__7h3wg50"
            viewBox="0 0 24 24"
          >
            <title>
              drive
            </title>
            <path
              d="M7.71,3.5L1.15,15L4.58,21L11.13,9.5M9.73,15L6.3,21H19.42L22.85,15M22.28,14L15.42,2H8.58L8.57,2L15.43,14H22.28Z"
            />
          </svg>
          <span
            class="style_tabItemLabel__1bi6wwc3"
          >
            Google Drive
          </span>
        </button>
      </li>
      <li
        class="style_tabItem__1bi6wwc1"
      >
        <button
          class="style_tabItemButton__1bi6wwc2"
          type="button"
        >
          <svg
            class="style_icon__7h3wg50"
            viewBox="0 0 24 24"
          >
            <title>
              setting
            </title>
            <path
              d="M12,15.5A3.5,3.5 0 0,1 8.5,12A3.5,3.5 0 0,1 12,8.5A3.5,3.5 0 0,1 15.5,12A3.5,3.5 0 0,1 12,15.5M19.43,12.97C19.47,12.65 19.5,12.33 19.5,12C19.5,11.67 19.47,11.34 19.43,11L21.54,9.37C21.73,9.22 21.78,8.95 21.66,8.73L19.66,5.27C19.54,5.05 19.27,4.96 19.05,5.05L16.56,6.05C16.04,5.66 15.5,5.32 14.87,5.07L14.5,2.42C14.46,2.18 14.25,2 14,2H10C9.75,2 9.54,2.18 9.5,2.42L9.13,5.07C8.5,5.32 7.96,5.66 7.44,6.05L4.95,5.05C4.73,4.96 4.46,5.05 4.34,5.27L2.34,8.73C2.21,8.95 2.27,9.22 2.46,9.37L4.57,11C4.53,11.34 4.5,11.67 4.5,12C4.5,12.33 4.53,12.65 4.57,12.97L2.46,14.63C2.27,14.78 2.21,15.05 2.34,15.27L4.34,18.73C4.46,18.95 4.73,19.03 4.95,18.95L7.44,17.94C7.96,18.34 8.5,18.68 9.13,18.93L9.5,21.58C9.54,21.82 9.75,22 10,22H14C14.25,22 14.46,21.82 14.5,21.58L14.87,18.93C15.5,18.67 16.04,18.34 16.56,17.94L19.05,18.95C19.27,19.03 19.54,18.95 19.66,18.73L21.66,15.27C21.78,15.05 21.73,14.78 21.54,14.63L19.43,12.97Z"
            />
          </svg>
          <span
            class="style_tabItemLabel__1bi6wwc3"
          >
            Settings
          </span>
        </button>
      </li>
      <li
        class="style_tabItem__1bi6wwc1"
      >
        <button
          class="style_tabItemButton__1bi6wwc2"
          type="button"
        >
          <svg
            class="style_icon__7h3wg50"
            viewBox="0 0 24 24"
          >
            <title>
              sign in
            </title>
            <path
              d="M10,17V14H3V10H10V7L15,12L10,17M10,2H19A2,2 0 0,1 21,4V20A2,2 0 0,1 19,22H10A2,2 0 0,1 8,20V18H10V20H19V4H10V6H8V4A2,2 0 0,1 10,2Z"
            />
          </svg>
          <span
            class="style_tabItemLabel__1bi6wwc3"
          >
            Sign In
          </span>
        </button>
      </li>
    </ul>
    <div
      class="style_content__1bi6wwc5"
    >
      <div
        class="style_playing__plyebw0"
      >
        <svg
          class="style_image__plyebw1 style_noImage__plyebw2"
          viewBox="0 0 100 100"
        >
          <title>
            no image
          </title>
          <path
            class="style_noImageBG__plyebw3"
            d="M0,0V100H100V0"
          />
          <text
            x="50%"
            y="25"
          >
            no image
          </text>
          <text
            x="50%"
            y="38"
          >
            available
          </text>
          <path
            d="M17,27L22,33L29,24L38,36H10M38,42A4,4 0 0,0 42,38V10A4,4 0 0,0 38,6H10A4,4 0 0,0 6,10V38A4,4 0 0,0 10,42"
            transform="translate(26 45)"
          />
        </svg>
        <dl
          class="style_info__plyebw4"
        >
          <dt>
            Album
          </dt>
          <dd />
          <dt>
            Album Artist
          </dt>
          <dd />
          <dt>
            Title
          </dt>
          <dd />
          <dt>
            Artist
          </dt>
          <dd />
          <dt>
            Disk
          </dt>
          <dd>
            
            
          </dd>
          <dt>
            Track
          </dt>
          <dd>
            
            
          </dd>
        </dl>
      </div>
    </div>
    <input
      class="style_seekBar__10xjpva2"
      max="0"
      min="0"
      type="range"
    />
    <div
      class="style_controller__10xjpva0"
    >
      <button
        class="style_iconButton__10xjpva1"
        type="button"
      >
        <svg
          class="style_icon__7h3wg50"
          viewBox="0 0 24 24"
        >
          <title>
            skip previous
          </title>
          <path
            d="M6,18V6H8V18H6M9.5,12L18,6V18L9.5,12Z"
          />
        </svg>
      </button>
      <button
        class="style_iconButton__10xjpva1"
        type="button"
      >
        <svg
          class="style_icon__7h3wg50"
          viewBox="0 0 24 24"
        >
          <title>
            play
          </title>
          <path
            d="M8,5.14V19.14L19,12.14L8,5.14Z"
          />
        </svg>
      </button>
      <button
        class="style_iconButton__10xjpva1"
        type="button"
      >
        <svg
          class="style_icon__7h3wg50"
          viewBox="0 0 24 24"
        >
          <title>
            skip next
          </title>
          <path
            d="M16,18H18V6H16M6,18L14.5,12L6,6V18Z"
          />
        </svg>
      </button>
      <span
        class="style_titleBox__10xjpva3"
      >
        <span
          class="style_marquee__r2bqb91 style_title__10xjpva4"
        >
          <span
            class="style_styleInner__r2bqb92"
          />
        </span>
        <span
          class="style_marquee__r2bqb91 style_artist__10xjpva5"
        >
          <span
            class="style_styleInner__r2bqb92"
          />
        </span>
      </span>
      <button
        class="style_iconButton__10xjpva1"
        type="button"
      >
        <svg
          class="style_icon__7h3wg50"
          viewBox="0 0 24 24"
        >
          <title>
            repeat off
          </title>
          <path
            d="M2,5.27L3.28,4L20,20.72L18.73,22L15.73,19H7V22L3,18L7,14V17H13.73L7,10.27V11H5V8.27L2,5.27M17,13H19V17.18L17,15.18V13M17,5V2L21,6L17,10V7H8.82L6.82,5H17Z"
          />
        </svg>
      </button>
      <button
        class="style_iconButton__10xjpva1"
        type="button"
      >
        <svg
          class="style_icon__7h3wg50"
          viewBox="0 0 24 24"
        >
          <title>
            shuffle off
          </title>
          <path
            d="M16,4.5V7H5V9H16V11.5L19.5,8M16,12.5V15H5V17H16V19.5L19.5,16"
          />
        </svg>
      </button>
      <span
        class="style_timeBox__10xjpva6"
      >
        00:00
        /
        00:00
      </span>
    </div>
  </div>
</div>
`;
